test/Microsoft.AspNetCore.Razor.Runtime.Test/Runtime/TagHelpers/TagHelperExecutionContextTest.cs

```diff
@@ -305,10 +305,10 @@ public void AllAttributes_IgnoresCase(string originalName, string updatedName)
         {
             // Arrange
             var executionContext = new TagHelperExecutionContext("p", tagMode: TagMode.StartTagAndEndTag);
-            executionContext.AllAttributes.Add(originalName, value: false);
+            executionContext.AddTagHelperAttribute(originalName, value: false);
 
             // Act
-            executionContext.AllAttributes[updatedName].Value = true;
+            executionContext.AllAttributes[updatedName] = true;
```

Comment:
We lose old in-place update but replacement should be explicit i.e. executionContext.AllAttributes[updatedName] = new TagHelperAttribute(updatedName, true)`.